### ISSUE TITLE
since 2.3.0 update, can't open a new container

### ISSUE THREAD

**Issue by danieljakots** [mozilla/testpilot-containers#566]
Hi,

I've been running using containers for a while and I really liked what I could with them (and it works fine). Since a few days ago I can't open a new container:
I click on the plugin in the toolbar I can see the list 'Personal' 'Shopping' 'Finance' and so on but when I click on one category the pop-up closes itself but it doesn't open a container. I tried to launch firefox from a terminal but it doesn't says anything.

I've another profile where it hasn't been updated and it's still 2.2.0 and there it works fine. I'm running firefox-53.0.3 on OpenBSD amd64 -current fwiw.

I hope it will get fixed soon as I really miss it :)

**Comment by jonathanKingston** [comment]
@danieljakots are you able to open the browser console.

1. Ctrl+Shift+J
2. Clear the console
3. Press the popup icon

**Comment by danieljakots** [comment]
Pasting what happens in the browser console:

Error in parsing value for ‘fill’.  Declaration dropped.  popup.css:390:8
Unknown property ‘enable-background’.  Declaration dropped.  container-sort.svg
Unknown property ‘enable-background’.  Declaration dropped.  container-add.svg
Unknown property ‘enable-background’.  Declaration dropped.  container-arrow.svg
Unknown property ‘enable-background’.  Declaration dropped.  container-hide.svg
Unknown property ‘enable-background’.  Declaration dropped. usercontext.svg
Unknown property ‘enable-background’.  Declaration dropped.  usercontext.svg
Unknown property ‘enable-background’.  Declaration dropped.  usercontext.svg
Unknown property ‘enable-background’.  Declaration dropped.  usercontext.svg
Unknown property ‘enable-background’.  Declaration dropped.  usercontext.svg
Unknown property ‘enable-background’.  Declaration dropped.

about:support

Application Basics
------------------

Name: Firefox
Version: 53.0.3
Build ID: 20170525005219
User Agent: Mozilla/5.0 (X11; OpenBSD amd64; rv:53.0) Gecko/20100101 Firefox/53.0
OS: OpenBSD 6.1
Multiprocess Windows: 0/1 (Disabled)
Safe Mode: false

Extensions
----------

Name: Application Update Service Helper
Version: 2.0
Enabled: true
ID: aushelper@mozilla.org

Name: Containers Experiment
Version: 2.3.0
Enabled: true
ID: @testpilot-containers

Name: HTTPS Everywhere
Version: 5.2.17
Enabled: true
ID: https-everywhere-eff@eff.org

Name: Multi-process staged rollout
Version: 1.14
Enabled: true
ID: e10srollout@mozilla.org

Name: Pocket
Version: 1.0.5
Enabled: true
ID: firefox@getpocket.com

Name: Privacy Badger
Version: 2017.5.9
Enabled: true
ID: jid1-MnnxcxisBPnSXQ-eff@jetpack

Name: Test Pilot
Version: 1.1.2-dev-c4639a2
Enabled: true
ID: @testpilot-addon

Name: uBlock Origin
Version: 1.12.4
Enabled: true
ID: uBlock0@raymondhill.net

Name: Web Compat
Version: 1.0
Enabled: true
ID: webcompat@mozilla.org

Name: JSONView
Version: 1.2.4
Enabled: false
ID: jsonview@brh.numbera.com

Name: NoScript
Version: 5.0.5
Enabled: false
ID: {73a6fe31-595d-460b-a920-fcc0f8843232}

Graphics
--------

Features
Compositing: Basic
Asynchronous Pan/Zoom: none
WebGL Renderer: Intel Open Source Technology Center -- Mesa DRI Intel(R) Haswell Desktop
WebGL2 Renderer: WebGL creation failed: * WebGL 2 requires support for the following features: transform_feedback2 * Exhausted GL driver options.
Audio Backend: sndio
GPU #1
Active: Yes
Description: Intel Open Source Technology Center -- Mesa DRI Intel(R) Haswell Desktop
Vendor ID: Intel Open Source Technology Center
Device ID: Mesa DRI Intel(R) Haswell Desktop
Driver Version: 3.0 Mesa 13.0.6

Diagnostics
AzureCanvasAccelerated: 0
AzureCanvasBackend: skia
AzureContentBackend: skia
AzureFallbackCanvasBackend: none
CairoUseXRender: 0
Decision Log
HW_COMPOSITING:
blocked by default: Acceleration blocked by platform
OPENGL_COMPOSITING:
unavailable by default: Hardware compositing is disabled




Important Modified Preferences
------------------------------

accessibility.typeaheadfind.flashBar: 0
browser.cache.disk.capacity: 358400
browser.cache.disk.filesystem_reported: 1
browser.cache.disk.hashstats_reported: 1
browser.cache.disk.smart_size.first_run: false
browser.cache.disk.smart_size.use_old_max: false
browser.cache.frecency_experiment: 1
browser.download.importedFromSqlite: true
browser.download.useDownloadDir: false
browser.places.smartBookmarksVersion: 8
browser.privatebrowsing.autostart: true
browser.search.useDBForOrder: true
browser.startup.homepage_override.buildID: 20170525005219
browser.startup.homepage_override.mstone: 53.0.3
browser.tabs.warnOnClose: false
browser.urlbar.suggest.history: false
browser.urlbar.suggest.openpage: false
dom.apps.lastUpdate.buildID: 20161111124649
dom.apps.lastUpdate.mstone: 49.0.2
dom.apps.reset-permissions: true
dom.max_chrome_script_run_time: 40
dom.max_script_run_time: 40
dom.mozApps.used: true
extensions.lastAppVersion: 53.0.3
font.internaluseonly.changed: false
font.size.fixed.x-western: 16
font.size.variable.x-western: 18
gfx.crash-guard.glcontext.appVersion: 45.0
gfx.crash-guard.glcontext.deviceID: Gallium 0.4 on AMD RV770 (DRM 2.29.0)
gfx.crash-guard.glcontext.driverVersion: 3.0 Mesa 11.0.9
gfx.crash-guard.status.glcontext: 2
media.autoplay.enabled: true
media.gmp-manager.buildID: 20170525005219
media.gmp-manager.lastCheck: 1496197250
media.gmp.storage.version.observed: 1
media.webrtc.debug.aec_log_dir: /tmp
media.webrtc.debug.log_file: /tmp/WebRTC.log
media.youtube-ua.override.to: 43
network.auth.allow-subresource-auth: 2
network.cookie.prefsMigrated: true
network.dns.disablePrefetch: true
network.http.speculative-parallel-limit: 0
network.predictor.cleaned-up: true
network.prefetch-next: false
network.protocol-handler.external.ftp: false
places.database.lastMaintenance: 1495680763
places.history.expiration.transient_current_max_pages: 122334
plugin.disable_full_page_plugin_for_types: application/pdf
print.print_bgcolor: false
print.print_bgimages: false
print.print_colorspace: default
print.print_downloadfonts: false
print.print_duplex: 0
print.print_evenpages: true
print.print_in_color: true
print.print_margin_bottom: 0.5
print.print_margin_left: 0.5
print.print_margin_right: 0.5
print.print_margin_top: 0.5
print.print_oddpages: true
print.print_orientation: 0
print.print_page_delay: 50
print.print_paper_data: 0
print.print_paper_height: 279.40
print.print_paper_name: na_letter
print.print_paper_size_type: 1
print.print_paper_size_unit: 1
print.print_paper_width: 215.90
print.print_plex_name: default
print.print_resolution_name: default
print.print_scaling: 1.00
print.print_shrink_to_fit: false
print.print_to_file: false
print.print_unwriteable_margin_bottom: 56
print.print_unwriteable_margin_left: 25
print.print_unwriteable_margin_right: 25
print.print_unwriteable_margin_top: 25
privacy.donottrackheader.enabled: true
privacy.donottrackheader.value: 1
privacy.sanitize.migrateClearSavedPwdsOnExit: true
privacy.sanitize.migrateFx3Prefs: true
privacy.usercontext.about_newtab_segregation.enabled: true
privacy.userContext.enabled: true
security.disable_button.openCertManager: false
services.sync.declinedEngines:
storage.vacuum.last.index: 1
storage.vacuum.last.places.sqlite: 1494383696

Important Locked Preferences
----------------------------

Places Database
---------------

JavaScript
----------

Incremental GC: true

Accessibility
-------------

Activated: false
Prevent Accessibility: 0

Library Versions
----------------

NSPR
Expected minimum version: 4.14
Version in use: 4.14

NSS
Expected minimum version: 3.30.2
Version in use: 3.30.2

NSSSMIME
Expected minimum version: 3.30.2
Version in use: 3.30.2

NSSSSL
Expected minimum version: 3.30.2
Version in use: 3.30.2

NSSUTIL
Expected minimum version: 3.30.2
Version in use: 3.30.2

Experimental Features
---------------------

**Comment by jonathanKingston** [comment]
I can't see anything obvious right now in any of that, especially as the item rendered. The only thing I can think of is it's a more severe #564 replication.

Could you try tabbing through the menu and using enter to see if this works?

**Comment by danieljakots** [comment]
If I understand correctly what you say: clicking on 'Containers' in the toolbar then pressing tab and enter to try to open a container, it's the same. The menu closes without opening the requested container.

**Comment by jonathanKingston** [comment]
Yeah that is what I meant.

- Could you provide a screenshot at all? (I know it will likely show nothing but worth a try)
- Do other browser actions become non interactive also, like pocket, testpilot, ublock origin, bookmark button etc?

Thanks for your patience and interest in Containers.

**Comment by jonathanKingston** [comment]
I was able to replicate similar in unbranded 53 on Ubuntu.

- Open containers browser action
- Click add button next to containers
- Container is created
- Go back to browser action and arrow is next to the container name
- Click arrow

Expected:
Panel to open

Actual:
Nothing happens

@kjozwiak would really appreciate your eyes on this as I dunno what is causing this and it might mean we have to cut an emergency release from the release branch.

**Comment by jonathanKingston** [comment]
Tested on latest release unbranded 53 on a fresh profile and the issue doesn't happen.

@bakulf have you seen this at all either?

**Comment by jonathanKingston** [comment]
Existing Containers profile, then updated to 2.3.0. Unbranded 53.0.2

> about:support

Application Update Service Helper	2.0	true	aushelper@mozilla.org
Containers Experiment	2.3.0	true	@testpilot-containers
Multi-process staged rollout	1.14	true	e10srollout@mozilla.org
Pocket	1.0.5	true	firefox@getpocket.com
Test Pilot	1.1.2-dev-f3fe9bf	true	@testpilot-addon
Web Compat	1.0	true	webcompat@mozilla.org
Ubuntu Modifications	3.2	false	ubufox@ubuntu.com

Stepping through the debugger it looks like an uncaptured promise here:
```
    return browser.runtime.sendMessage({
      method: "getTabs",
      userContextId: Logic.userContextId(identity.cookieStoreId),
    }).then(this.buildInfoTable);
```

I also get errors like this:
```
console.trace: 
_fronts/inspector.js 770 WalkerFront<.getMutations</<
_/Promise-backend.js 917 process
_/Promise-backend.js 801 walkerLoop
_/Promise-backend.js 737 scheduleWalkerLoop/<

console.error: 
  TypeError: can't access dead object: CssLogic.getComputedStyle@resource://gre/modules/commonjs/toolkit/loader.js -> resource://devtools/server/css-logic.js:686:7
get computedStyle@resource://gre/modules/commonjs/toolkit/loader.js -> resource://devtools/server/actors/inspector.js:360:12
get isDisplayed@resource://gre/modules/commonjs/toolkit/loader.js -> resource://devtools/server/actors/inspector.js:375:9
_onReflows@resource://gre/modules/commonjs/toolkit/loader.js -> resource://devtools/server/actors/inspector.js:1027:11
emit@resource://gre/modules/commonjs/toolkit/loader.js -> resource://devtools/shared/event-emitter.js:191:13
_startEventLoop@resource://gre/modules/commonjs/toolkit/loader.js -> resource://devtools/server/actors/reflow.js:314:7
setTimeout_timer@resource://gre/modules/Timer.jsm:30:5
```

**Comment by jonathanKingston** [comment]
Debugging
- index.js `async getTabs(args) {`
- Which then calls: `sendReply(this[message.method](message));`
- returns a promise which I was expecting to be resolved here:
  - http://searchfox.org/mozilla-central/source/toolkit/components/extensions/ExtensionChild.jsm#352


Unanswered questions:
- Is there anything different between unbranded and release
- Why would only a stale profile cause this
- Is there anything different in the bundled build (Have not been able to reproduce on the production branch)

**Comment by jonathanKingston** [comment]
IRC paraphrase for anyone else interested about the "fix":

This fixes the panel opening issue, it creates the likelihood that some icons might turn into pages even for nightly.
In nightly currently new tab and other icons go missing anyway which is another fix I did elsewhere... this fixes them all to be pages if they don't load

The main thing I was testing was the panel that lists the containers tabs, I don't think there is anywhere else that code is active.
In Release you will see goat icons and a 300ms delay and in Nightly you will see the icons and new goat icons for new tab pages.
I was only able to replicate using a broken old profile that had it's container install pre 2.3.0 and direct from the store.
I wasn't able to replicate by building the experiment/firefox and creating the mockup.
As Baku mentioned the way to "test" a broken profile is back it up and edit the XPI with vim to contain the patch

**Comment by jonathanKingston** [comment]
The follow up to this bug should be using web extensions which likely wouldn't be impacted by this issue at all here.
Pushing a bigger SDK removal patch to prod late Thurs or Friday seems better than lying about what favicon the tab has and a 300ms delay.

**Comment by jonathanKingston** [comment]
As mentioned with the bug author @danieljakots, this actually is a Private browsing issue. This makes it a duplicate of #469. However the STR mentioned in https://github.com/mozilla/testpilot-containers/issues/566#issuecomment-305444672 is what the real issue I was debugging.

Perhaps we should change the title of this bug to reflect that, I don't really want to lift and shift all the comments to just close this bug.

**Comment by danieljakots** [comment]
Yes confirmed, everything works fine now. Thanks a lot for walking me through debugging the problem!
(You can close the issue if you want, I don't do it myself as I'm not sure if you want to keep it around)

**Comment by kjozwiak** [comment]
> @kjozwiak would really appreciate your eyes on this as I dunno what is causing this and it might mean we have to cut an emergency release from the release branch.

I've been trying to reproduce this for a few hours on multiple platforms without any luck :/

> Perhaps we should change the title of this bug to reflect that, I don't really want to lift and shift all the comments to just close this bug.

I was going to suggest spinning off your [issue](https://github.com/mozilla/testpilot-containers/issues/566#issuecomment-305444672) into a seperate bug. Following this gets pretty confusing especially as there's two bugs here. One that @danieljakots reported and the other one that @jonathanKingston found [here](https://github.com/mozilla/testpilot-containers/issues/566#issuecomment-305444672). Someone who doesn't know the history behind this bug might assume all the debugging/comments are referring to the original problem which they're not..

**Comment by jonathanKingston** [comment]
Closed as I have moved away the bug comments of a different issue to #574
